bashflux/cmd/channels.go

```diff
@@ -0,0 +1,179 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/mainflux/mainflux/clients"
+	"github.com/spf13/cobra"
+)
+
+var chanEndPoint = "channels"
+
+// Channels
+var CmdChannels = &cobra.Command{
+	Use:   "channels",
+	Short: "Manipulation with channels",
+	Long:  `Manipulation with channels: create, delete or update channels`,
+	Run: func(cmdCobra *cobra.Command, args []string) {
+		if len(args) == 1 {
+			GetChannels(args[0])
+		} else {
+			LogUsage(cmdCobra.Short)
+		}
+	},
+}
+
+var CmdCreateChannel = &cobra.Command{
+	Use:   "create",
+	Short: "create <JSON_channel> <user_auth_token>",
+	Long:  `Creates new channel and generates it's UUID`,
+	Run: func(cmdCobra *cobra.Command, args []string) {
+		if len(args) == 2 {
+			msg := args[0]
+			token := args[1]
+			CreateChannel(msg, token)
+		} else {
+			LogUsage(cmdCobra.Short)
+		}
+	},
+}
+
+var CmdGetChannel = &cobra.Command{
+	Use:   "get",
+	Short: "get <user_auth_token> or get <channel_id> <user_auth_token>",
+	Long:  `Gets list of all channels or gets channel by id`,
+	Run: func(cmdCobra *cobra.Command, args []string) {
+		if len(args) == 1 {
+			GetChannels(args[0])
+		} else if len(args) == 2 {
+			GetChannel(args[0], args[1])
+		} else {
+			LogUsage(cmdCobra.Short)
+		}
+	},
+}
+
+var CmdUpdateChannel = &cobra.Command{
+	Use:   "update",
+	Short: "update <channel_id> <JSON_string> <user_auth_token>",
+	Long:  `Updates channel record`,
+	Run: func(cmdCobra *cobra.Command, args []string) {
+		if len(args) == 3 {
+			UpdateChannel(args[0], args[1], args[2])
+		} else {
+			LogUsage(cmdCobra.Short)
+		}
+	},
+}
+
+var CmdDeleteChannel = &cobra.Command{
+	Use:   "delete",
+	Short: "delete <channel_id> <user_auth_token>",
+	Long:  `Delete channel by ID`,
+	Run: func(cmdCobra *cobra.Command, args []string) {
+		if len(args) == 2 {
+			if args[0] == "all" {
+				DeleteAllChannels(args[1])
+			} else {
+				DeleteChannel(args[0], args[1])
+			}
+		} else {
+			LogUsage(cmdCobra.Short)
+		}
+	},
+}
+
+// CreateChannel - creates new channel and generates UUID
+func CreateChannel(msg string, token string) {
```

Comment:
Make a package-private API client (until we provide one from the services), and bind these methods to it. Later on use them as you're doing it right now.